File: C_Programming/Day_7/program61.c
// Description : Write a program that accepts array size , target  , elements from the user ,  
//               & returns the frequency of that target element from the array.

#include<stdio.h>
#include<stdlib.h>

int CountFrequency(int *Arr , int iSize , int iTarget)
{
    int iCnt = 0;
    int iCount = 0;    

    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        if(Arr[iCnt] == iTarget)
        {
            iCount++;
        }
    }

    return iCount;
    
}

void Display(int * Arr , int iSize)
{
    int iCnt = 0;

    printf("Elements from the array are : \n");
    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        printf("%d\t",Arr[iCnt]);
    }
    printf("\n");
}

void Accept(int * Arr , int iSize)
{
    int iCnt = 0;

    printf("Enter the elements : \n");
    for(iCnt = 0 ; iCnt < iSize ; iCnt++)
    {
        scanf("%d",&Arr[iCnt]);
    }
}

int main()
{
    int * Arr = NULL;
    int iLength = 0;
    int iRet = 0;
    int iTarget = 0;

    printf("Enter the size of the array : ");
    scanf("%d",&iLength);

    printf("Enter the number that you want to serach in the array : ");
    scanf("%d",&iTarget);

    Arr = (int *)malloc(sizeof(int) * iLength);

    Accept(Arr,iLength);
    Display(Arr,iLength);

    iRet = CountFrequency(Arr , iLength , iTarget);
    printf("Frequency is : %d\n",iRet);
    

    return 0;
}

File: Python_Programming/Day_7/program61.py
# Description:  Write a program that accepts array size , target  , elements from the user ,  
#               & returns the frequency of that target element from the array.

def Accept(Arr , Size):
    Value = 0

    print("Enter the elements : ")
    for i in range(Size):
      Value = int(input())
      Arr.append(Value) 

def Display(Arr , Size):

    print("Elements are : ")
    for i in range(Size):
        print(Arr[i] , "\t" , end = "")
    
    print("\n")

def CountFrequency(Arr , Size , Target):
    Count = 0

    for i in range(Size):
        if(Arr[i] == Target):
            Count += 1

    return Count

def main():
    Arr = list()
    Length = 0
    Ret = 0
    Target = 0

    print("Enter the size of the list : ",end = "")
    Length = int(input())

    print("Enter the number that you want to search : " , end = "")
    Target = int(input())

    Accept(Arr , Length)
    Display(Arr,Length)

    Ret = CountFrequency(Arr , Length , Target)
    print(f"Frequency of {Target} is : ",Ret)
    

if __name__ == "__main__":
    main()